train.rb

```diff
@@ -1 +1,61 @@
+class Train
+  attr_reader :number, :type, :count_wagons, :speed
 
+  def initialize(number, type, count_wagons)
+    @number = number
+    @type = type
+    @count_wagons = count_wagons
+    @speed = 0
+  end
+
+  def decrease_speed(value)
+    @speed -= value
+    @speed = 0 if @speed.negative?
+  end
+
+  def increase_speed(value)
+    @speed += value
+  end
+
+  def add_wagon
+    @count_wagons += 1 if @speed.zero?
+  end
+  
+  def remove_wagon
+    @count_wagons -= 1 if @speed.zero? && @count_wagons.positive?
+  end
+
+  def accept_route(route)
+    @route = route
+    @route.stations.first.add_train(self)
+    @current_station = 0
```

Comment:
лучше чтобы избежать путанницы дать имя, которое бы явно давало понять, что тут не объект класса Стейшен, а индекс этого объекта
```suggestion
    @current_station_index = 0
```